pagopa/io-app
# [IOBP-110, IOBP-114] new wallet payment methods cards

## Short description
added new design Wallet payment method cards:
- small
- big
- carousel

| small | big+carousel |
|--------|--------|
| <img width="200" alt="Screenshot 2023-07-10 at 12 14 37" src="https://github.com/pagopa/io-app/assets/60693085/c33a49e4-d42d-4b64-a099-c1ad9d2a4838"> | <img width="200" alt="Screenshot 2023-07-10 at 12 14 55" src="https://github.com/pagopa/io-app/assets/60693085/3e8f84dc-bfad-411b-b2ef-5d31018a352d"> | 

## List of changes proposed in this pull request
- added required components
- added skeletons for said components
- added error state for small cards
- added new `LogoPaymentOrDefaultIcon` utility component, with its own jsdoc reference
- moved new payment cards in their own folders in the ui components
- refactored `useListItemBaseSpringAnimation` to be a now generic `useBaseSpringItemAnimation`
- added necessary entry page in the design system
- added required tests

## How to test
run the required tests, also navigate to the Design System page (dev mode active -> profile -> scroll down to design system -> cards) and make sure that nothing has rendering problems both with and without the beta DS flag
